test/jdk/java/nio/channels/FileChannel/BlockDeviceSize.java

```diff
@@ -34,31 +34,40 @@
 import java.nio.channels.FileChannel;
 import java.nio.file.AccessDeniedException;
 import java.nio.file.NoSuchFileException;
+import java.util.List;
+
 import static java.nio.file.StandardOpenOption.*;
 
 
 public class BlockDeviceSize {
-    private static final String BLK_FNAME = "/dev/sda1";
-    private static final Path BLK_PATH = Paths.get(BLK_FNAME);
+    private static final List<String> BLK_FNAMES = List.of("/dev/sda1", "/dev/nvme0n1") ;
+
+    public static Path getBlkPath(String blkFileName) {
+        return Paths.get(blkFileName);
+    }
 
     public static void main(String[] args) throws Throwable {
-        try (FileChannel ch = FileChannel.open(BLK_PATH, READ);
-             RandomAccessFile file = new RandomAccessFile(BLK_FNAME, "r")) {
+        for (String blkFname: BLK_FNAMES) {
+            Path blkPath = getBlkPath(blkFname);
+            try (FileChannel ch = FileChannel.open(blkPath, READ);
+                 RandomAccessFile file = new RandomAccessFile(blkFname, "r")) {
+
+                long size1 = ch.size();
+                long size2 = file.length();
+                if (size1 != size2) {
```

Comment:
A passing comment on this test is that it just checks that FC::size and RAF::size return the same size. We are sitting on prototype changes that reimplement RAF on FC so it won't be comparing two different implementations if that goes ahead. It will at least test that the size method doesn't throw.

Reply:
Ok, so if I'm understanding well your comment, it will still make sense to keep this test even if it's only as a validation that the `.size` call won't fail, while the `size1 != size2` will become irrelevant when the changes that you're mentioning get merged.

Reply:
That's right, the best we can do then is to check that size is >= 0.

Reply:
Perfect, I'll leave the check there, and we can change it to `size() >= 0` when the reimplementation of RAF on FC is done.

Reply:
The summary in the test description is "Block devices should not report size=0 on Linux". This test will actually pass if the block device size is reported as 0. So I think the test summary needs to change or the test checks that the size is > 0.

Reply:
You're correct. I believe that until the prototype changes are merged we can keep the equality tests and add a `!= 0` test.

See: https://github.com/openjdk/jdk/pull/19021/commits/e7fd93af9815d1643122d99135df3ac75e6a939c